actions/promote_package.yaml

```diff
@@ -0,0 +1,40 @@
+---
+name: promote_package
+runner_type: mistral-v2
+description: Promote a package to production from staging.
+enabled: true
+entry_point: workflows/promote_package.yaml
+parameters:
+  package:
+    type: string
+    description: Package to promote
+    required: true
+  version:
+    type: string
+    description: Version to promote
+    required: true
+  revision:
+    type: string
+    description: Revision to promote
+    required: true
+  os:
+    description: OS to promote (will promote all if not specified)
+    required: true
+    default: all
```

Comment:
Packages for every platform/os has independent revision discovery and can have unsynced revision number.

In fact, we had this already and it's normal case, because for example: 
- changing number of parallel builds in CircleCI to 1 will generate packages for 1 platform, rather then 5. It means revision is incremented only for that platform/os.
- triggering new build via CircleCI API can generate packages for specific platform. See ChatOps: https://github.com/StackStorm/st2-packages/issues/147

All these cases will bring revision number unsync between different OSes.
So it could be a problem here in future when promoting all platforms and assuming they have same revision number.

-----

^^ Just FYI to consider. Though it's not a big issue for the moment, however it will definitely bite in future.